examples/3d/chess.rs

```diff
@@ -0,0 +1,482 @@
+use bevy::prelude::*;
+
+fn main() {
+    App::new()
+        .insert_resource(Msaa { samples: 4 })
+        .insert_resource(WindowDescriptor {
+            title: "Chess!".to_string(),
+            width: 1280.0,
+            height: 720.0,
+            ..Default::default()
+        })
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup)
+        .add_startup_system(create_board)
+        .add_startup_system(create_white_pieces)
+        .add_startup_system(create_black_pieces)
+        .run();
+}
+
+fn setup(mut commands: Commands) {
+    commands.spawn_bundle(PerspectiveCameraBundle {
+        transform: Transform::from_xyz(0.0, 5.0, 8.).looking_at(Vec3::ZERO, Vec3::Y),
+        ..Default::default()
+    });
+    commands.spawn_bundle(PointLightBundle {
+        transform: Transform::from_translation(Vec3::new(30.0, 30.0, 30.0)),
+        point_light: PointLight {
+            intensity: 600000.,
+            range: 100.,
+            ..Default::default()
+        },
+        ..Default::default()
+    });
+}
+
+fn create_board(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    let mesh = meshes.add(Mesh::from(shape::Plane { size: 1.0 }));
+    let white_material = materials.add(Color::rgb(1.0, 0.9, 0.9).into());
+    let black_material = materials.add(Color::rgb(0.0, 0.1, 0.1).into());
+
+    // Add 64 Squares
+    for i in 0..8 {
+        for j in 0..8 {
+            commands.spawn_bundle(PbrBundle {
+                mesh: mesh.clone(),
+                material: if (i + j + 1) % 2 == 0 {
```

Comment:
This is a common and clever trick. In teaching code, I would have a comment here explaining why it works.

Reply:
I actually learned this trick over a decade ago.. 😂 .. I can't think of a comment which would make sense in the current context outside of re-iterating it in mathematics. Any recommendations?

Reply:
So, start by describing the intent: to create an alternating pattern of grid cells.

If you start at a cell, it is either black or white. If you move in any direction along either x or y, you want to change color.

When you increase or decrease either x or y by 1, the parity of your modulus flips. This is the same as the rule for the colors, so we can create a mapping between 0 and black and 1 and white.

(Also this code is clearer if you remove the +1 and just swap the bindings).